jobserver/releases.py

```diff
@@ -1,16 +1,29 @@
 import hashlib
+from datetime import timedelta
 from pathlib import Path
 
 from django.db import transaction
+from django.utils import timezone
 
 from .models import Release, ReleaseFile
 from .models.outputs import absolute_file_path
+from .signing import AuthToken
 
 
 class ReleaseFileAlreadyExists(Exception):
     pass
 
 
+def build_auth_token(*, url, username, expiry=None, backend_token):
```

Comment:
maybe it's worth abstracting this to a backend and workspace, rather than a url? so sth like:
 
```
def build_auth_token(backend, workspace, user, expiry=None):
    ...
    f = furl(backend.level_4_url)
    f.path.segments += ["workspace", workspace.name]

    token = AuthToken(url=f.url, user=user.username, expiry=expiry)

    return token.sign(key=backend.backend_token, salt="hatch")
```

This means the url generation is abstracted out, and we can change it as needed without the view needing to care.